features/filesystem/fat/FATMisc.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2006-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file contains miscellaneous functionality used by the FAT filesystem interface code.
+ */
+
+#include "FATMisc.h"
+#include <errno.h>
+#include "platform/retarget.h"
+
+
+/* @brief   Set errno based on the error code returned from underlying filesystem
+ *
+ * @param   res     result returned from underlying filesystem
+ *
+ * @return  No return value
+ */
+void FATFileSystemSetErrno(FRESULT res)
```

Comment:
According to style guide ([here](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/cont/code_style/)), functions should be in lower-case snake case: `fat_file_system_set_errno`